cartography/intel/gcp/sql.py

```diff
@@ -0,0 +1,236 @@
+        :param project_id: Current Google Project Id
+
+        :type gcp_update_tag: timestamp
+        :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
+    """
+    ingest_sql_users = """
+    UNWIND {sql_users} as user
+    MERGE(u:GCPSQLUser{id:user.id})
+    ON CREATE SET
+        u.firstseen = timestamp()
+    SET
+        u.name = user.name,
+        u.host = user.host,
+        u.instance = user.instance,
+        u.project = user.project,
+        u.type = user.type
+    WITH user, u
+    MATCH (owner:GCPProject{id:{ProjectId}})-[r1:Resource]->(i:GCPSQLInstance{id:instance.id})
+    MERGE (owner)-[r1:RESOURCE]->(i)-[r2:uses]<-(u)
+    ON CREATE SET
+        r1.firstseen = timestamp,
+        r2.firstseen = timestamp,
+        r1.lastupdated = {gcp_update_tag},
+        r2.lastupdated = {gcp_update_tag}
+    """
+    neo4j_session.run(
+        ingest_sql_users,
+        sql_users = sql_users,
+        ProjectId=project_id,
+        gcp_update_tag=gcp_update_tag,
+    )
+
+@timeit
+def cleanup_sql_instances(neo4j_session: neo4j.Session, common_job_parameters: Dict) -> None:
+    """
+    Delete out-of-date GCP SQL Instances and relationships
+
+    :type neo4j_session: The Neo4j session object
+    :param neo4j_session: The Neo4j session
+
+    :type common_job_parameters: dict
+    :param common_job_parameters: Dictionary of other job parameters to pass to Neo4j
+
+    :rtype: NoneType
+    :return: Nothing
+    """
+    run_cleanup_job('gcp_sql_cleanup.json', neo4j_session, common_job_parameters)
+
+@timeit
+def sync(
+    neo4j_session: neo4j.Session, sql: Resource, project_id: str, gcp_update_tag: int,
+    common_job_parameters: Dict
+) -> None:
+    """
+    Get GCP Cloud SQL Instances and Users using the Cloud Function resource object, ingest to Neo4j, and clean up old data.
+
+    :type neo4j_session: The Neo4j session object
+    :param neo4j_session: The Neo4j session
+
+    :type sql: The GCP Cloud SQL resource object created by googleapiclient.discovery.build()
+    :param sql: The GCP Cloud SQL resource object
+
+    :type project_id: str
+    :param project_id: The project ID of the corresponding project
+
+    :type gcp_update_tag: timestamp
+    :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
+
+    :type common_job_parameters: dict
+    :param common_job_parameters: Dictionary of other job parameters to pass to Neo4j
+
+    :rtype: NoneType
+    :return: Nothing
+    """
+    logger.info("Syncing GCP Cloud SQL for project %s.", project_id)
+    #INSTANCES
+    instances =  get_sql_instances(sql,project_id)
+    load_sql_instances(instances,project_id,gcp_update_tag)
+    #SQL USERS
+    users = get_sql_users(sql,project_id)
+    load_sql_users(users,project_id,gcp_update_tag)
+    # TODO scope the cleanup to the current project - https://github.com/lyft/cartography/issues/381
+    cleanup_sql_instances(neo4j_session, common_job_parameters)
```

Comment:
this cleanup should follow right after load.

Reply:
The cleanup json also cleans the users. So wouldn't it be better to place the cleanup after loading the users also.